### PR TITLE
Lint the repo

### PR DESCRIPTION
**Pull request by maxrjones**
This PR runs the pre-commit linters against all files in the repo except the intro which is addressed in #30.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,21 +29,20 @@ Washington, DC 20001, United States
...
-|:-----------:|:-------------------------------------------------------:|
-| 08:30-08:40 | Introduction to GMT / PyGMT                             |
-| 08:40-08:55 | Installation                                            |
-| 08:55-09:45 | Tutorial 1 - First figure + Subplots / layout           |
-| 09:45-10:15 | **Break**                                               |
+| Time (EST)  |                                     Topic                                     |
+| :---------: | :---------------------------------------------------------------------------: |
+| 08:30-08:40 |                          Introduction to GMT / PyGMT                          |
+| 08:40-08:55 |                                 Installation                                  |
+| 08:55-09:45 |                 Tutorial 1 - First figure + Subplots / layout                 |
```
Comment (yvonnefroehlich) [inline README.md]: To be consistent with the titles of the tutorials (sorry for destroying the linting...).

```suggestion
| 08:55-09:45 |                Tutorial 1 - First figure and Subplots / layout               |
```

File: README.md [hunk 2]
```diff
@@ -27,21 +29,20 @@ Washington, DC 20001, United States
...
 | 10:15-11:00 | Tutorial 2 - Integration with scientific Python ecosystem: pandas / GeoPandas |
-| 11:00-11:45 | Tutorial 3 - Integration with scientific Python ecosystem: Xarray (grids)     |
-| 11:45-12:45 | **Lunch**                                               |
-| 12:45-13:30 | Tutorial 4 - Geophysics (Seismology)                    |
-| 13:30-14:15 | Tutorial 5 - 3-D Topography (Planetary / Antarctic maps) |
-| 14:15-14:45 | **Break**                                               |
-| 14:45-15:30 | Tutorial 6 - Animations with GMT                        |
-| 15:30-17:00 | Final exercises / mini-project                          |
-
+| 11:00-11:45 |   Tutorial 3 - Integration with scientific Python ecosystem: Xarray (grids)   |
```
Comment (yvonnefroehlich) [inline README.md]:
```suggestion
| 11:00-11:45 |   Tutorial 3 - Integration with scientific Python ecosystem: Xarray (gridded data)   |
```

File: README.md [hunk 3]
```diff
@@ -27,21 +29,20 @@ Washington, DC 20001, United States
...
-| 08:40-08:55 | Installation                                            |
-| 08:55-09:45 | Tutorial 1 - First figure + Subplots / layout           |
-| 09:45-10:15 | **Break**                                               |
+| Time (EST)  |                                     Topic                                     |
+| :---------: | :---------------------------------------------------------------------------: |
+| 08:30-08:40 |                          Introduction to GMT / PyGMT                          |
+| 08:40-08:55 |                                 Installation                                  |
+| 08:55-09:45 |                 Tutorial 1 - First figure + Subplots / layout                 |
+| 09:45-10:15 |                                   **Break**                                   |
 | 10:15-11:00 | Tutorial 2 - Integration with scientific Python ecosystem: pandas / GeoPandas |
```
Comment (yvonnefroehlich) [inline README.md]:
```suggestion
| 10:15-11:00 | Tutorial 2 - Integration with scientific Python ecosystem: pandas / GeoPandas (tabular data) |
```

**Review by yvonnefroehlich — COMMENTED** [review]
Thanks @maxrjones for doing this PR 🙂 !

File: book/tut01_firstfigure.ipynb [hunk 4]
```diff
@@ -249,16 +245,15 @@
...
+    "# and this way can save grid (outgrid) as a file in your folder.\n",
     "# https://www.generic-mapping-tools.org/remote-datasets/\n",
     "\n",
-    "# pygmt.grdcut(grid=\"@earth_relief_03m\",  \n",
-    "#              region=region,\n",
-    "#              outgrid=\"Alaska.grd\", \n",
-    "#              ) "
+    "# pygmt.grdcut(\n",
+    "#    grid=\"@earth_relief_03m\", region=region, outgrid=\"Alaska.grd\"\n",
+    "# )"
```
Comment (yvonnefroehlich) [inline book/tut01_firstfigure.ipynb:256]: See commit https://github.com/GenericMappingTools/agu24workshop/pull/31/commits/fa256ec6f968ed8dd1448041716d9b517ff5670c.

**Comment by weiji14** [comment]
Noticed that the auto-generated `conda-lock.yml` file was also linted, which is a rather big diff... I hope that we don't need to update any dependencies anymore, but if so, I'll make a mental note to run `conda-lock lock` followed by `pre-commit`.

**Pull request merged**